todoman/ui.py

```diff
@@ -74,9 +96,16 @@ def edit(self):
         Shows the UI for editing a given todo. Returns True if modifications
         were saved.
         """
-        loop = urwid.MainLoop(self._ui, unhandled_input=self._keypress,
-                              handle_mouse=False)
-        loop.run()
+        self._loop = urwid.MainLoop(self._ui, unhandled_input=self._keypress,
+                                    handle_mouse=False)
+        try:
+            self._loop.run()
+        except Exception:
+            try:  # Try to leave terminal in usable state
+                self._loop.stop()
```

Comment:
This was taken from khal, where I also added it. The urwid docs don't indicate it's necessary, however, I found https://github.com/urwid/urwid/issues/77